pkg/webui/console/lib/events/utils.js

```diff
@@ -23,13 +23,29 @@ export const defineSyntheticEvent = name => data => ({
   data,
 })
 
-export const createSyntheticEventFromError = error => {
+const convertError = error => {
   if (error instanceof Error) {
-    const errorString = error.toString()
-    if (error.message === 'network error' || error.message === 'Error in body stream') {
-      return createNetworkErrorEvent({ error: errorString })
+    return {
+      ...error,
+      message: error.message,
+      name: error.name,
+      // The stack is omitted intentionally, as it is not relevant for a user.
     }
+  }
+  return error
+}
 
-    return createUnknownErrorEvent({ error: errorString })
+export const createSyntheticEventFromError = error => {
+  if (error instanceof Error) {
+    if (
+      error.name === 'ConnectionError' ||
+      error.name === 'ConnectionClosedError' ||
+      error.name === 'ConnectionTimeoutError'
+    ) {
+      return createNetworkErrorEvent({ error: convertError(error) })
+    } else if (error.name === 'ProtocolError') {
```

Comment:
Protocol errors are basically Console implementation errors at this point, and maybe should be sent to Sentry or have a dedicated event, hence the specific branch.